proxy_agent/src/proxy/authorization_rules.rs

```diff
@@ -0,0 +1,331 @@
+            mode: "disabled".to_string(),
+            roles: Some(vec![Role {
+                name: "test".to_string(),
+                privileges: vec!["test".to_string(), "test1".to_string()],
+            }]),
+            privileges: Some(vec![
+                Privilege {
+                    name: "test".to_string(),
+                    path: "/test".to_string(),
+                    queryParameters: None,
+                }
+            ]),
+            identities: Some(vec![
+                Identity {
+                    name: "test".to_string(),
+                    exePath: Some("test".to_string()),
+                    groupName:Some("test".to_string()) ,
+                    processName: Some("test".to_string()),
+                    userName: Some("test".to_string()), 
+                }
+            ]),
+            roleAssignments: Some(vec![RoleAssignment {
+                role: "test".to_string(),
+                identities: vec!["test".to_string()],
+            }]),
+            id: "0".to_string(),
+        };
+        let rules = AuthorizationRules::from_authorization_item(authorization_item);
+        assert_eq!(rules.defaultAllowed, false);
+        assert_eq!(rules.mode, "disabled");
+        assert_eq!(rules.rules.is_some(), true);
+
+        let url = url::Url::parse("http://localhost/test?").unwrap();
+        assert_eq!(rules.is_allowed(0, url.to_string(), claims.clone()), true);
+
+        // Test enforce mode, identity not match
+        let authorization_item: AuthorizationItem = AuthorizationItem {
+            defaultAccess: "deny".to_string(),
+            mode: "enforce".to_string(),
+            roles: Some(vec![Role {
+                name: "test".to_string(),
+                privileges: vec!["test".to_string(), "test1".to_string()],
+            }]),
+            privileges: Some(vec![
+                Privilege {
+                    name: "test".to_string(),
+                    path: "/test".to_string(),
+                    queryParameters: None,
+                }
+            ]),
+            identities: Some(vec![
+                Identity {
+                    name: "test1".to_string(),
+                    exePath: Some("test".to_string()),
+                    groupName:Some("test".to_string()) ,
+                    processName: Some("test".to_string()),
+                    userName: Some("test".to_string()), 
+                }
+            ]),
+            roleAssignments: Some(vec![RoleAssignment {
+                role: "test".to_string(),
+                identities: vec!["test1".to_string()],
+            }]),
+            id: "0".to_string(),
+        };
+        let rules = AuthorizationRules::from_authorization_item(authorization_item);
+        assert_eq!(rules.defaultAllowed, false);
+        assert_eq!(rules.mode, "enforce");
+        assert_eq!(rules.rules.is_some(), true);
+        
+        let url = url::Url::parse("http://localhost/test?").unwrap();
```

Comment:
can we try use "/test?" as url

Reply:
no because match url::Url::parse in is_allowed requires a full path URL